File: 3461-find-the-minimum-area-to-cover-all-ones-i/find-the-minimum-area-to-cover-all-ones-i.cpp
class Solution {
public:
    int minimumArea(vector<vector<int>>& grid) {
        int x1 = INT_MAX, y1 = INT_MAX, x2 = 0, y2 = 0;

        for(int i = 0; i < grid.size(); i++){
            for(int j = 0; j < grid[0].size(); j++){
                if(grid[i][j] == 1){
                    x1 = min(x1, i);
                    x2 = max(x2, i);
                    y1 = min(y1, j);
                    y2 = max(y2, j);
                }
            }
        }
        return (x2 - x1 + 1) * (y2 - y1 + 1);
    }
};

File: 3461-find-the-minimum-area-to-cover-all-ones-i/find-the-minimum-area-to-cover-all-ones-i.py
class Solution(object):
    def minimumArea(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        x1 = y1 = float('inf')
        x2 = y2 = float('-inf')
        found = False

        for i, row in enumerate(grid):
            for j, val in enumerate(row):
                if val == 1:
                    found = True
                    x1 = min(x1, i)
                    x2 = max(x2, i)
                    y1 = min(y1, j)
                    y2 = max(y2, j)

        if not found: return 0
        return (x2 - x1 + 1) * (y2 - y1 + 1)

